src/basis_extend.jl

```diff
@@ -0,0 +1,112 @@
+import ITensors: pause
+#
+# Possible improvements
+#  - allow a maxdim argument to be passed to `extend`
+#    and through `basis_extend`
+#  - current behavior is letting bond dimension get too
+#    big when used in imaginary time evolution
+#  - come up with better names:
+#    > should `basis_extend` be called `krylov_extend`?
+#    > should `extend` be called `basis_extend`?
+#  - Use (1-tau*H)|psi> to generate "Krylov" vectors
+#    instead of H|psi>. Needed?
+#
+
+"""
+Given an MPS psi and a collection of MPS phis,
+returns an MPS which is equal to psi
+(has fidelity 1.0 with psi) but whose MPS basis
+is extended to contain a portion of the basis of
+the phis that is orthogonal to the MPS basis of psi.
+"""
+function extend(psi::MPS, phis::Vector{MPS}; kwargs...)
```

Comment:
Agreed that `basis_extend` would be a good name to match what is used in the paper, this would be an internal function anyway.

Reply:
Or maybe `extend_basis`.